admin/code/LeftAndMain.php

```diff
@@ -773,6 +773,29 @@ public function SiteTreeAsUL() {
 	}
 
 	/**
+	 * Gets the current search filter for this request, if available
+	 *
+	 * @throws InvalidArgumentException
+	 * @return LeftAndMain_SearchFilter
+	 */
+	protected function getSearchFilter() {
+		// Check for given FilterClass
+		$params = $this->getRequest()->getVar('q');
+		if(empty($params['FilterClass'])) {
+			return null;
+		}
+
+		// Validate classname
+		$filterClass = $params['FilterClass'];
+		$filterInfo = new ReflectionClass($filterClass);
+		if(!$filterInfo->implementsInterface('LeftAndMain_SearchFilter')) {
```

Comment:
`is_subclass_of($params['FilterClass'], 'LeftAndMain_SearchFilter')` might be more efficient.